### PR TITLE
Fix some gcc warnings

### PR DESCRIPTION
**Pull request by YarikTH**

### CODE REVIEW

File: v8pp/class.ipp [hunk 1]
```diff
@@ -375,6 +375,7 @@ V8PP_IMPL classes* classes::instance(operation op, v8::Isolate* isolate)
 			delete info;
 			isolate->SetData(V8PP_ISOLATE_DATA_SLOT, nullptr);
 		}
+		return nullptr;
```
Comment (YarikTH) [inline v8pp/class.ipp:378]:
```
v8pp/class.ipp:373:3:
  warning: this statement may fall through [-Wimplicit-fallthrough=]

    373 |   if (info)
        |   ^~

v8pp/class.ipp:378:2:
  note: here

    378 |  default:
        |  ^~~~~~~
```

File: v8pp/json.ipp [hunk 2]
```diff
@@ -64,7 +64,8 @@ V8PP_IMPL v8::Local<v8::Object> json_object(v8::Isolate* isolate, v8::Local<v8::
 			{
 				value = v8::JSON::Stringify(context, value).FromMaybe(v8::String::Empty(isolate));
 			}
-			result->Set(context, name, value);
+			v8::Maybe<bool> res = result->Set(context, name, value);
+			(void)res;
```
Comment (YarikTH) [inline v8pp/json.ipp:68]:
```
v8pp/json.ipp:67:36:
  warning: ignoring return value of ‘v8::Maybe<bool>
  v8::Object::Set(v8::Local<v8::Context>, v8::Local<v8::Value>,
  v8::Local<v8::Value>)’, declared with attribute warn_unused_result
  [-Wunused-result]

     67 |    result->Set(context, name, value);
        |                                    ^

/usr/include/v8/v8.h:3247:37: note: declared here

   3247 |   V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
        |                                     ^~~
```

File: v8pp/utility.hpp [hunk 3]
```diff
@@ -213,7 +213,7 @@ struct is_sequence<T, void_t<typename T::value_type,
 template<typename T, typename U = void>
 struct has_reserve : std::false_type
 {
-	static void reserve(T& container, size_t capacity) {} // no-op
+	static void reserve(T& /*container*/, size_t /*capacity*/) {} // no-op
```
Comment (YarikTH) [inline v8pp/utility.hpp:216]:
```
v8pp/utility.hpp:216:25: warning: unused parameter ‘container’ [-Wunused-parameter]
  216 |  static void reserve(T& container, size_t capacity) {} // no-op
      |                      ~~~^~~~~~~~~
v8pp/utility.hpp:216:43: warning: unused parameter ‘capacity’ [-Wunused-parameter]
  216 |  static void reserve(T& container, size_t capacity) {} // no-op
      |                                    ~~~~~~~^~~~~~~~
```

File: v8pp/utility.hpp [hunk 4]
```diff
@@ -232,10 +232,10 @@ struct has_reserve<T, void_t<decltype(std::declval<T>().reserve(0))>> : std::tru
 template<typename T>
 struct is_array : std::false_type
 {
-	static void check_length(size_t length) {} // no-op for non-arrays
+	static void check_length(size_t /*length*/) {} // no-op for non-arrays
```
Comment (YarikTH) [inline v8pp/utility.hpp:235]:
```
v8pp/utility.hpp:235:34: warning: unused parameter ‘length’ [-Wunused-parameter]
  235 |  static void check_length(size_t length) {} // no-op for non-arrays
      |                           ~~~~~~~^~~~~~
```

File: v8pp/utility.hpp [hunk 5]
```diff
@@ -232,10 +232,10 @@ struct has_reserve<T, void_t<decltype(std::declval<T>().reserve(0))>> : std::tru
...
 
 	template<typename U>
-	static void set_element_at(T& container, size_t index, U&& item)
+	static void set_element_at(T& container, size_t /*index*/, U&& item)
```
Comment (YarikTH) [inline v8pp/utility.hpp:238]:
```
v8pp/utility.hpp:238:50: warning: unused parameter ‘index’ [-Wunused-parameter]
  238 |  static void set_element_at(T& container, size_t index, U&& item)
      |                                           ~~~~~~~^~~~~
```

**Comment by pmed** [comment]
I've actually fixed some warnings earlier in commit c145f120eda9bbedd6b44cc263aad36a292731cf
Pushed it just now.

**Comment by YarikTH** [comment]
It seems that your commit addresses all the cases that I handled here, but mostly in a different ways. Never heard about the comment "// fallthrough". Good to know.

**Pull request closed without merging**